src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,296 @@
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.lib.team3015.subsystem.FaultReporter;
+import frc.lib.team3061.leds.LEDs;
+import frc.lib.team3061.leds.LEDs.IntakeLEDState;
+
+import java.util.ArrayList;
+
+import org.littletonrobotics.junction.Logger;
+
+public class Intake extends SubsystemBase {
+
+  private static final String SUBSYSTEM_NAME = "INTAKE";
+
+  private final IntakeIO io;
+  private final IntakeIOInputsAutoLogged inputs = new IntakeIOInputsAutoLogged();
+  private final LEDs leds;
+
+  private final IntakeMotor[] intakeMotors = { 
+    IntakeMotor.RIGHT_ROLLER, 
+    IntakeMotor.KICKER,
+  };
+
+  enum IntakeMotor {
+    RIGHT_ROLLER,
+    KICKER
+  }
+
+  private int intakeAndKickerTimeout;
+
+  // Intakes, Kicker, and Shooter (the second kicker IR sensor)
+  enum IntakeState {
+    EMPTY,
+    NOTE_IN_INTAKE,
+    NOTE_IN_INTAKE_AND_KICKER,
+    NOTE_IN_KICKER,
+    NOTE_IN_KICKER_AND_SHOOTER,
+    NOTE_IN_SHOOTER
+  }
+
+  private IntakeState intakeState;
+  private IntakeState mostRecentIntakeState;
+  private boolean checkComplete;
+
+  private boolean manualOverrideEnabled;
+
+  public Intake(IntakeIO io) {
+    this.io = io;
+    leds = LEDs.getInstance();
+    intakeState = IntakeState.EMPTY;
+    mostRecentIntakeState = null;
+    checkComplete = false;
+
+    leds.setIntakeLEDState(IntakeLEDState.WAITING_FOR_GAME_PIECE);
+    this.intakeGamePiece();
+    
+    FaultReporter.getInstance().registerSystemCheck(SUBSYSTEM_NAME, getSystemCheckCommand());
+  }
+
+  @Override
+  public void periodic() {
+    io.updateInputs(inputs);
+    Logger.processInputs("Intake", inputs);
+    Logger.recordOutput("Intake/State", intakeState.toString());
+
+    this.runIntakeStateMachine();
+  }
+
+  public void runIntakeStateMachine() {
+
+    if (intakeState == IntakeState.EMPTY) {
+      // if it is empty, then this is the only 1 possible next steps / situations
+      //   1. we are truly empty, and waiting for a game piece
+      if (inputs.isRollerIRBlocked) {
+        intakeState = IntakeState.NOTE_IN_INTAKE;
+        leds.setIntakeLEDState(IntakeLEDState.HAS_GAME_PIECE);
+        this.transitionGamePiece();
+      } else if (inputs.isShooterIRBlocked) {
+        intakeState = IntakeState.NOTE_IN_KICKER;
+        this.repelGamePiece();
+      }
+    } else if (intakeState == IntakeState.NOTE_IN_INTAKE) {
+      if (inputs.isKickerIRBlocked) {
+        intakeState = IntakeState.NOTE_IN_INTAKE_AND_KICKER;
+        intakeAndKickerTimeout = 0;
+        this.transitionGamePiece();
+      }
+    } else if (intakeState == IntakeState.NOTE_IN_INTAKE_AND_KICKER) {
+      intakeAndKickerTimeout++;
+
+      if (!inputs.isRollerIRBlocked) {
+        intakeState = IntakeState.NOTE_IN_KICKER;
+        this.transitionGamePiece();
+        this.repelGamePiece();
+      } else if (intakeAndKickerTimeout > IntakeConstants.IN_BETWEEN_TIMEOUT_SECONDS * 50) {
```

Comment:
I don't think we need the timeout anymore since we don't have any dead zones with the IR sensors.